pkg/election/election.go

```diff
@@ -267,18 +267,14 @@ func (e *Election) campaignLoop(ctx context.Context, session *concurrency.Sessio
 
 			err2 := elec.Campaign(ctx2, e.infoStr)
 			if err2 != nil {
-				// sometimes we noticed auto Resign does not work, so we try to manually Resign.
-				e.l.Debug("before manually resign",
-					zap.String("current election key", elec.Key()),
-					zap.Any("current election header", elec.Header()))
-				resignCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-				defer cancel()
-				if err3 := elec.Resign(resignCtx); err3 != nil {
-					e.l.Warn("failed to manually resign", zap.Error(err3))
+				// because inner commit may return undetermined error, we try to delete the election key manually
+				deleted, err3 := e.ClearSessionIfNeeded(ctx, e.ID())
```

Comment:
Can you please tell me why I need to delete this key manually ?

and in old logic, we do `Resign` if `Campaign` failed, after this change ,why don't we do this anymore?

Reply:
see the issue in PR description